RecoLocalCalo/EcalRecProducers/plugins/EcalUncalibRecHitWorkerWeights.cc

```diff
@@ -92,9 +94,15 @@ EcalUncalibRecHitWorkerWeights::run( const edm::Event & evt,
         }
 */
         if (detid.subdetId()==EcalEndcap) {
-                result.push_back(uncalibMaker_endcap_.makeRecHit(*itdg, pedVec, pedRMSVec, gainRatios, weights, testbeamEEShape));
+                EcalUncalibratedRecHit rhit = (uncalibMaker_endcap_.makeRecHit(*itdg, pedVec, pedRMSVec, gainRatios, weights, testbeamEEShape));
+                if( ((EcalDataFrame)(*itdg)).hasSwitchToGain6()  ) rhit.setFlagBit( EcalUncalibratedRecHit::kHasSwitchToGain6 );
+	            if( ((EcalDataFrame)(*itdg)).hasSwitchToGain1()  ) rhit.setFlagBit( EcalUncalibratedRecHit::kHasSwitchToGain1 );
+                result.push_back(rhit);
```

Comment:
emplace_back is more appropriate, although the compiler will likely optimize to the same result